File: frontend/src/lib/inputLimits.ts
// Keep in sync with backend: Aureus.UseCases/Validation/InputLimits.cs
export const InputLimits = {
  emailMaxLength: 254,
  passwordMaxLength: 128,

  workspaceNameMaxLength: 120,
  categoryNameMaxLength: 120,
  accountNameMaxLength: 120,

  transactionNameMaxLength: 200,
  transactionNoteMaxLength: 500,
  insightsQuestionMaxLength: 500,
} as const;

File: backend/src/Aureus.UseCases/Validation/InputLimits.cs
namespace Aureus.UseCases.Validation;

internal static class InputLimits
{
    internal const int EmailMaxLength = 254;
    internal const int PasswordMaxLength = 128;

    internal const int WorkspaceNameMaxLength = 120;
    internal const int CategoryNameMaxLength = 120;
    internal const int AccountNameMaxLength = 120;

    internal const int TransactionNameMaxLength = 200;
    internal const int TransactionNoteMaxLength = 500;
    internal const long TransactionMaxAmountMinor = 100_000_000_000L;
}
